client/src/controllers/LoaderController.test.js

```diff
@@ -0,0 +1,99 @@
+import { Application } from '@hotwired/stimulus';
+import { LoaderController } from './LoaderController';
+
+describe('LoaderController', () => {
+  beforeEach(() => {
+    document.body.innerHTML = `
+            <button
+                type="submit"
+                class="button button-longrunning"
+                data-controller="w-loader"
+                data-w-loader-active-class="button-longrunning-active"
+                data-w-loader-active-value="Signing in…"
+                data-w-loader-duration-seconds-value="40"
+                data-action="w-loader#activate"
+            >
+                <svg>...</svg>
+                <em data-w-loader-target="label" id="em-el">Sign in</em>
+            </button>
+        `;
+    Application.start().register('w-loader', LoaderController);
+  });
+
+  afterEach(() => {
+    document.body.innerHTML = '';
+    jest.clearAllMocks();
+    jest.clearAllTimers();
+  });
+
+  it('should not execute the rest of the code if the form is not valid', () => {
+    const form = document.createElement('form');
```

Comment:
I recommend you just put an actual form wrapper element inside your html - it's simpler to reason about and avoids all this extra mock functions below. 

Then if you create a field in the html that won't be valid - you don't have to mock the validation function.